File: 974. Subarray Sums Divisible by K/n.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        res, running_sum = 0, 0
        count = [0] * k
        count[0] = 1
        for num in nums:
            running_sum += num
            mod = running_sum % k
            if mod < 0:
                mod += k
            res += count[mod]
            count[mod] += 1
        return res

File: 974. Subarray Sums Divisible by K/n.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        int running_sum = 0, res = 0;
        vector<int> count (k, 0);
        count[0] = 1;
        for (auto num: nums) {
            running_sum += num;
            int mod = running_sum % k;
            if (mod < 0)
                mod += k;
            res += count[mod];
            count[mod] += 1;
        }
        return res;
    }
};